### PR TITLE
smt2_parser: parse Array sort

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 09ad8fe3e37980b09085bd260efbde4d74f8c45c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87880341
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 32b9e2c379ae303ce7925a63d0c2fa9f6409fc99).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87880689

File: src/solvers/smt2/smt2_parser.cpp [hunk 1]
```diff
@@ -975,6 +975,38 @@ typet smt2_parsert::sort()
...
+      }
+
+      // we can turn arrays that map a bitvector type to something else
+      // into our 'array_typet'
+      if(domain.id() == ID_unsignedbv)
+      {
+        const auto size = to_unsignedbv_type(domain).largest_expr();
+        return array_typet(range, size);
+      }
+      else if(domain.id() == ID_signedbv)
```
Comment (martin-cs) [inline src/solvers/smt2/smt2_parser.cpp]: What would generate these?  Seeing as SMT-LIB doesn't have a signed bit-vector sort and the BitVec is being treated as unsigned?

**Review by martin-cs — APPROVED** [review]

File: src/solvers/smt2/smt2_parser.cpp (hunk 1 above)
Comment (kroening) [reply src/solvers/smt2/smt2_parser.cpp]: Indeed, should not happen.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 0cf070be63f8f0791126c371b42bd3ff2949eab8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87998797
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Pull request merged**